File: SDKs/googleapis.com/firebasedatabase/v1beta/go/pkg/models/operations/firebasedatabaseprojectslocationsinstancesundelete.go
package operations

import (
	"openapi/pkg/models/shared"
)

type FirebasedatabaseProjectsLocationsInstancesUndeletePathParams struct {
	Name string `pathParam:"style=simple,explode=false,name=name"`
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteQueryParams struct {
	DollarXgafv    *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
	AccessToken    *string           `queryParam:"style=form,explode=true,name=access_token"`
	Alt            *shared.AltEnum   `queryParam:"style=form,explode=true,name=alt"`
	Callback       *string           `queryParam:"style=form,explode=true,name=callback"`
	Fields         *string           `queryParam:"style=form,explode=true,name=fields"`
	Key            *string           `queryParam:"style=form,explode=true,name=key"`
	OauthToken     *string           `queryParam:"style=form,explode=true,name=oauth_token"`
	PrettyPrint    *bool             `queryParam:"style=form,explode=true,name=prettyPrint"`
	QuotaUser      *string           `queryParam:"style=form,explode=true,name=quotaUser"`
	UploadType     *string           `queryParam:"style=form,explode=true,name=uploadType"`
	UploadProtocol *string           `queryParam:"style=form,explode=true,name=upload_protocol"`
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption1 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption2 struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteSecurity struct {
	Option1 *FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption1 `security:"option"`
	Option2 *FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption2 `security:"option"`
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteRequest struct {
	PathParams  FirebasedatabaseProjectsLocationsInstancesUndeletePathParams
	QueryParams FirebasedatabaseProjectsLocationsInstancesUndeleteQueryParams
	Request     map[string]interface{} `request:"mediaType=application/json"`
	Security    FirebasedatabaseProjectsLocationsInstancesUndeleteSecurity
}

type FirebasedatabaseProjectsLocationsInstancesUndeleteResponse struct {
	ContentType      string
	DatabaseInstance *shared.DatabaseInstance
	StatusCode       int64
}

File: SDKs/googleapis.com/firebasedatabase/v1beta/typescript/src/sdk/models/operations/firebasedatabaseprojectslocationsinstancesundelete.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class FirebasedatabaseProjectsLocationsInstancesUndeletePathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption1 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption2 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, option=true" })
  option1?: FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption1;

  @Metadata({ data: "security, option=true" })
  option2?: FirebasedatabaseProjectsLocationsInstancesUndeleteSecurityOption2;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: FirebasedatabaseProjectsLocationsInstancesUndeletePathParams;

  @Metadata()
  queryParams: FirebasedatabaseProjectsLocationsInstancesUndeleteQueryParams;

  @Metadata({ data: "request, media_type=application/json" })
  request?: Map<string, any>;

  @Metadata()
  security: FirebasedatabaseProjectsLocationsInstancesUndeleteSecurity;
}


export class FirebasedatabaseProjectsLocationsInstancesUndeleteResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  databaseInstance?: shared.DatabaseInstance;

  @Metadata()
  statusCode: number;
}
